### PR TITLE
fix: Date Time shows incorrect date on UI

### PR DESCRIPTION
**Pull request by omkarK06**
## Summary by CodeRabbit

- **New Features**
	- Improved handling of logs tab lifecycle and error messages for better user experience.
	- Enhanced responsiveness of table cell styling based on state changes.
	- Introduced utility function for validating date and time strings to enhance error handling.
	- Added new enumeration and interface to track UI component states related to logs and stream exploration.

- **Bug Fixes**
	- Adjusted error message display logic for clarity.
	- Refined timezone selection handling to ensure correct initial settings.

- **Refactor**
	- Reorganized component methods for better maintainability and readability.
	- Transitioned table cell class management to a reactive reference for dynamic updates.
	- Minor adjustments to the DateTime component for improved code clarity and functionality.

### CODE REVIEW

File: web/src/plugins/logs/TenstackTable.vue [hunk 1]
```diff
@@ -546,6 +540,21 @@ const hasDefaultSourceColumn = computed(
...
+    tableCellClass.value = [
+      hasDefaultSourceColumn.value && !props.wrap
+        ? "tw-table-cell"
+        : "tw-block",
+      !props.wrap && "tw-overflow-hidden tw-text-ellipsis tw-whitespace-nowrap",
+      props.wrap && "tw-break-words",
+    ];
+  },
+);
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/TenstackTable.vue]: Computed didn't worked here, so converted it to watcher

File: web/src/plugins/logs/Index.vue [hunk 2]
```diff
@@ -541,6 +536,8 @@ export default defineComponent({
     const showSearchHistory = ref(false);
     let parser: any;
 
+    const isLogsMounted = ref(false);
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:543]: Activated should not run for the first time.

File: web/src/components/DateTime.vue [hunk 3]
```diff
@@ -752,7 +751,7 @@
 
         const startTimeStamp = date.subtractFromDate(
           endTimeStamp,
-          JSON.parse(subtractObject)
+          JSON.parse(subtractObject),
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/DateTime.vue]: Done

File: web/src/components/DateTime.vue [hunk 4]
```diff
@@ -878,7 +877,7 @@
       update(() => {
         const value = val.toLowerCase();
         filteredOptions = options.filter(
-          (column: any) => column.toLowerCase().indexOf(value) > -1
+          (column: any) => column.toLowerCase().indexOf(value) > -1,
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/DateTime.vue]: Done

File: web/src/components/DateTime.vue [hunk 5]
```diff
@@ -773,15 +772,15 @@ export default defineComponent({
           start = new Date();
         } else {
           start = new Date(
-            selectedDate.value.from + " " + selectedTime.value.startTime
+            selectedDate.value.from + " " + selectedTime.value.startTime,
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/DateTime.vue]: Done

File: web/src/plugins/logs/Index.vue [hunk 6]
```diff
@@ -770,6 +696,163 @@ export default defineComponent({
...
+      loadLogsData();
+
+      if (isCloudEnvironment()) {
+        setupCloudSpecificThreshold();
+      }
+
+      searchObj.meta.quickMode = isQuickModeEnabled();
+
+      isLogsMounted.value = true;
+    }
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue]: Done

File: web/src/plugins/logs/Index.vue [hunk 7]
```diff
@@ -767,6 +693,163 @@
...
+        searchObj.organizationIdentifier !==
+        store.state.selectedOrganization.identifier
+      );
+    }
+
+    // Helper function for handling the visualize tab
+    function handleVisualizeTab() {
+      handleRunQueryFn();
+    }
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue]: Done

File: web/src/plugins/logs/Index.vue [hunk 8]
```diff
@@ -767,6 +693,163 @@ export default defineComponent({
...
+    // Setup cloud-specific organization threshold
+    function setupCloudSpecificThreshold() {
+      MainLayoutCloudMixin.setup().getOrganizationThreshold(store);
+    }
+
+    // Helper function to check if quick mode is enabled
+    function isQuickModeEnabled() {
+      return store.state.zoConfig.quick_mode_enabled;
+    }
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue]: Done

File: web/src/components/DateTime.vue [hunk 9]
```diff
@@ -735,6 +732,10 @@ export default defineComponent({
       return periodMapping[relativePeriod.value];
     });
 
+    function isValidDateTimeString(dateStr: string, timeStr: string): boolean {
+      return !isNaN(Date.parse(`${dateStr} ${timeStr}`));
+    }
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/DateTime.vue:737]: Fine for now

File: web/src/plugins/logs/Index.vue [hunk 10]
```diff
@@ -544,6 +540,8 @@ export default defineComponent({
     const showSearchHistory = ref(false);
     let parser: any;
 
+    const isLogsMounted = ref(false);
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:544]: Error handling is done in handleActivation method

File: web/src/plugins/logs/Index.vue [hunk 11]
```diff
@@ -737,6 +664,13 @@
           searchObj.data.stream.streamType =
             router.currentRoute.value.query.stream_type;
           resetSearchObj();
+
+          // As when redirecting from search history to logs page, date type was getting set as absolute, so forcefully keeping it relative.
+          searchBarRef.value.dateTimeRef.setRelativeTime(
+            router.currentRoute.value.query.period,
+          );
+          searchObj.data.datetime.type = "relative";
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/Index.vue:673]: Works for now

**Review by bjp232004 — APPROVED** [review]

**Pull request merged**